sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -151,9 +156,10 @@ class Analyzer(
 
   lazy val batches: Seq[Batch] = Seq(
     Batch("Hints", fixedPoint,
-      new ResolveHints.ResolveJoinStrategyHints(conf),
-      ResolveHints.ResolveCoalesceHints,
-      new ResolveHints.RemoveAllHints(conf)),
+      new ResolveHints.ResolveJoinStrategyHints(conf) +:
+        ResolveHints.ResolveCoalesceHints +:
+        extendedResolutionHints :+
```

Comment:
Oh, I see. 

@gatorsmile and @cloud-fan . How do you think about this extension?

Reply:
cc: @maryannxue , too. (I think this kind of extensions is basically bug-prone and we always need a better design...)

Reply:
@maropu @maryannxue  Any advise to improve this feature.